### ISSUE TITLE
isset(...) usage inspection's null-comparison suggestion is not intelligent

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#1117]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive
| Plugin         | Php Inspections (EA Extended), 3.0.8.1 |
| Language level | e.g. PHP 7.1                         |

Current behaviour (description/screenshot:)
--
Given the following code:
```php
        if (0) {
            $x = 0;
        }

        $y = isset($x);
```

PHPEA suggests that `isset($x)` be replaced by `$x !== null`, but `$x` is defined conditionally, so making this replacement results in a PHP notice: `Undefined variable: x`.

Expected behaviour
--
Replacing the `isset()` usage as PHPEA recommends is strictly incorrect when the variable is conditionally defined. PHPEA should not be making recommendations that are incorrect. Moreover, making the replacement PHPEA recommends triggers the built-in "Undefined variable" inspections for `$x`, so it should be possible for PHPEA to detect this scenario and avoid making the invalid recommendation.

Environment details
---
2018.2.5
